sdcm/tester.py

```diff
@@ -1257,106 +1246,171 @@ def _check_build_started():
-            columns = list(zip(result.column_names, [typelist.typename for typelist in result.column_types]))
+            self.log.debug('Rows in the {} MV before saving: {}'.format(src_table, len(source_table_rows)))
+
             insert_statement = session.prepare(
-                'insert into {keyspace}.{name} ({columns}) values ({values})'.format(keyspace=dest_keyspace,
-                                                                                     name=dest_table,
-                                                                                     columns=', '.join(
-                                                                                         [c[0] for c in columns]),
-                                                                                     values=', '.join(['?' for _ in columns])))
-
-            for row in result.current_rows:
-                session.execute(insert_statement, row)
-            source_rows_count = len(result.current_rows)
-
-            result = session.execute("SELECT * FROM {keyspace}.{name}".format(keyspace=dest_keyspace, name=dest_table))
-            if len(result.current_rows) != source_rows_count:
-                self. log.warning('Problem during copying data. '
-                                  'Rows in source table: {source}; '
-                                  'Rows in destination table: {destination}.'.format(source=source_rows_count,
-                                                                                     destination=len(result.current_rows))
-                                  )
-                return False
-            self.log.debug('All rows have been copied from {from_name} to {to_name}'.format(from_name=src_table,
-                                                                                            to_name=dest_table))
-        self.log.debug('Finish copying data')
+                'insert into {keyspace}.{name} ({columns}) '
+                'values ({values})'.format(keyspace=dest_keyspace,
+                                           name=dest_table,
+                                           columns=', '.join(columns),
+                                           values=', '.join(['?' for _ in columns])))
+
+            # Save all rows
+            # Workers = Parallel queries = (nodes in cluster) ✕ (cores in node) ✕ 3
+            # (from https://www.scylladb.com/2017/02/13/efficient-full-table-scans-with-scylla-1-6/)
+            cores = self.db_cluster.nodes[0].cpu_cores
+            if not cores:
+                # If CPU core didn't find, put 8 as default
+                cores = 8
+            max_workers = len(self.db_cluster.nodes) * cores * 3
+
+            session.default_consistency_level = ConsistencyLevel.QUORUM
+
+            execute_params = [(session, insert_statement, row) for row in source_table_rows]
+            ParallelObject(execute_params, num_workers=max_workers, timeout=3600, disable_logging=True).run(
```

Comment:
why ddin't you use `execute_concurrent`: https://docs.datastax.com/en/developer/python-driver/3.22/api/cassandra/concurrent/

Reply:
also please look on `execute_concurrent_with_args`

Reply:
Done